include/helib/EncryptedArray.h

```diff
@@ -1431,11 +1435,18 @@ class EncryptedArrayDerived<PA_cx> : public EncryptedArrayBase
    * @param ctxt Ciphertext to decrypt.
    * @param sKey Secret key to be used for decryption.
    * @param ptxt Plaintext into which to decrypt.
+   * @param prec `CKKS` precision to be used (must be defaulted if Scheme is `BGV`).
    * Decrypt a `Ctxt` ciphertext object to a `Ptxt` plaintext one relative to
    * a specific scheme.
    **/
+
+  // VJS-FIXME: something seems odd here. This code clearly only
+  // works for CKKS because ptxtArray has type vector<cx_double>.
+  // However, Scheme could be CKKS or BGV.  Is there a specialized
+  // version of this somewhere?
+  // TODO: document this better (especially the prec parameter)
   template <typename Scheme>
-  void decrypt(const Ctxt& ctxt, const SecKey& sKey, Ptxt<Scheme>& ptxt) const
+  void decrypt(const Ctxt& ctxt, const SecKey& sKey, Ptxt<Scheme>& ptxt, OptLong prec = OptLong()) const
```

Comment:
```suggestion
  void decrypt(const Ctxt& ctxt, const SecKey& sKey, Ptxt<Scheme>& ptxt) const
```

Removed `prec` parameter to avoid compilation error for now, until implementation is completed.